server/src/main/java/io/druid/server/coordinator/DruidCoordinatorRuntimeParams.java

```diff
@@ -303,9 +304,30 @@ public Builder withSegmentReplicantLookup(SegmentReplicantLookup lookup)
       return this;
     }
 
-    public Builder withDatasources(Collection<ImmutableDruidDataSource> dataSourcesCollection)
+    public Builder withDataSources(Map<String, VersionedIntervalTimeline<String, DataSegment>> dataSources)
     {
-      dataSources.addAll(Collections.unmodifiableCollection(dataSourcesCollection));
+      this.dataSources = dataSources;
+      return this;
+    }
+
+    public Builder withDataSources(Collection<ImmutableDruidDataSource> dataSourcesCollection)
+    {
+      dataSourcesCollection.forEach(
+          dataSource -> {
+            VersionedIntervalTimeline<String, DataSegment> timeline = dataSources.computeIfAbsent(
+                dataSource.getName(),
+                k -> new VersionedIntervalTimeline<>(String.CASE_INSENSITIVE_ORDER)
```

Comment:
@jihoonson why `CASE_INSENSITIVE_ORDER`, whereas `DruidCoordinatorCleanupOvershadowed` and `DruidCoordinatorRuleRunner` use natural order for the same thing?

Reply:
Hmm I don't remember, but it looks wrong. It would be better to call `VersionedIntervalTimeline.forSegments`.

Reply:
How do you know that `CASE_INSENSITIVE_ORDER` is wrong? Why natural order is right?

Reply:
I meant it's wrong in terms of what other places do. But, it shouldn't cause any real bug or error because currently version is a string representation of `DateTime`.